editor/editor/mode-switcher.js

```diff
@@ -0,0 +1,54 @@
+class ModeSwitcher extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.toggle = this.toggle.bind( this );
+		this.state = {
+			opened: false
+		};
+	}
+
+	toggle( event ) {
+		event.preventDefault();
+		this.setState( {
+			opened: ! this.state.opened
+		} );
+	}
+
+	render() {
+		const { opened } = this.state;
+		const modes = [
+			{ value: 'visual', label: 'Visual' },
+			{ value: 'text', label: 'Text' },
+		];
+		const switchMode = ( mode ) => () => {
+			this.setState( { opened: false } );
+			this.props.onSwitch( mode );
+		};
+		const currentMode = modes.find( ( { value } ) => value === this.props.mode );
+
+		return (
+			<div className="editor-mode-switcher">
+				<button
+					className="editor-mode-switcher__toggle"
+					onClick={ this.toggle }
+					aria-label="Switch the editor mode"
```

Comment:
We should probably start thinking about localization sooner than later. I might focus on this today (for a separate pull request).